apps/fhir/bluebutton/tests/test_read.py

```diff
@@ -61,3 +65,104 @@ def test_fhir_conformance_filter(self, mock_get_resource_names):
+    @patch('apps.dot_ext.throttling.TokenRateThrottle.get_rate')
+    def test_read_throttle(self,
+                           mock_rates):
+        mock_rates.return_value = '1/day'
+        # create the user
+        first_access_token = self.create_token('John', 'Smith')
+
+        response = self.client.get(
+            reverse(
+                'bb_oauth_fhir_read_or_update_or_delete',
+                kwargs={
+                    'resource_type': 'Patient',
+                    'resource_id': 12}),
+            Authorization="Bearer %s" % (first_access_token))
+
+        self.assertEqual(response.status_code, 403)
+
+        self.assertTrue(response.has_header("X-RateLimit-Limit"))
+        self.assertEqual(response.get("X-RateLimit-Limit"), "1")
+
+        self.assertTrue(response.has_header("X-RateLimit-Remaining"))
+        self.assertEqual(response.get("X-RateLimit-Remaining"), "0")
+
+        self.assertTrue(response.has_header("X-RateLimit-Reset"))
+        # 86400.0 is 24 hours
+        self.assertEqual(response.get("X-RateLimit-Reset"), '86400.0')
+
+        response = self.client.get(
+            reverse(
+                'bb_oauth_fhir_read_or_update_or_delete',
+                kwargs={
+                    'resource_type': 'Patient',
+                    'resource_id': 12}),
+            Authorization="Bearer %s" % (first_access_token))
+
+        self.assertEqual(response.status_code, 429)
+        # Assert that the proper headers are in place
+        self.assertTrue(response.has_header("X-RateLimit-Limit"))
+        self.assertEqual(response.get("X-RateLimit-Limit"), "1")
+
+        self.assertTrue(response.has_header("X-RateLimit-Remaining"))
+        self.assertEqual(response.get("X-RateLimit-Remaining"), "0")
+
+        self.assertTrue(response.has_header("X-RateLimit-Reset"))
+        # 86400.0 is 24 hours
+        self.assertTrue(float(response.get("X-RateLimit-Reset")) < 86400.0)
+
+        self.assertTrue(response.has_header("Retry-After"))
+        self.assertEqual(response.get("Retry-After"), "86400")
+
+        # Assert that the search endpoint is also ratelimited
+        response = self.client.get(
+            reverse(
+                'bb_oauth_fhir_search',
+                kwargs={
+                    'resource_type': 'Patient'}),
+            Authorization="Bearer %s" % (first_access_token))
+
+        self.assertEqual(response.status_code, 429)
+
+        # Assert that another token is not rate limited
+        second_access_token = self.create_token('Bob', 'Bobbington')
+        self.assertFalse(second_access_token == first_access_token)
+
+        response = self.client.get(
+            reverse(
+                'bb_oauth_fhir_read_or_update_or_delete',
+                kwargs={
+                    'resource_type': 'Patient',
+                    'resource_id': 12}),
+            Authorization="Bearer %s" % (second_access_token))
+
+        self.assertEqual(response.status_code, 403)
```

Comment:
My only concern on this test is that maybe we should use a valid token that will return a 200 response? Just to make sure the check is happening in the right order. I realize that the 403 check happens later right now, but that could theoretically change in the future.

Reply:
I figured the 403 was happening because the test env didn't have a connection to the backend servivce. Does the travis job have the right configs so that it can actually talk to the backend, if so what username/password should I setup so it'll return a 200? (cuz that would be great!)